code/crates/test/tests/unit/certificates/commit.rs

```diff
@@ -17,81 +19,97 @@ impl CertificateBuilder for Commit {
         CommitCertificate::new(height, round, value_id, votes)
     }
 
-    fn verify_certificate(
+    async fn verify_certificate(
         ctx: &TestContext,
         signer: &Ed25519Provider,
         certificate: &Self::Certificate,
         validator_set: &ValidatorSet,
         threshold_params: ThresholdParams,
     ) -> Result<(), CertificateError<TestContext>> {
-        signer.verify_commit_certificate(ctx, certificate, validator_set, threshold_params)
+        signer
```

Comment:
Instead of making everything in the tests async, can we instead use `futures::executor::block_on` to run the future? This is guaranteed to work as the test signing provider is purely synchronous and does not involve any async runtime whatsoever.